### ISSUE TITLE
Why didn't you use grep? 

### ISSUE THREAD

**Issue by GregEremeev** [soveran/hi#1]
It covers all your cases listed in the README file.
I was just curious.

**Comment by soveran** [comment]
Hey @GregEremeev, that's a good question because the README doesn't mention why this tool was created. I was helping a company that needed to find exact matches in 80 GB text files, and `grep` would crash on their platform. As a workaround, they were trying to stream the contents of the file with an XML library in Ruby, which would take ages to process the file and it would segfault ocassionally. So I gave them two tools, an XML streaming tool written in C++ and this one. I suggested they just use `hi`, because for their use case and exact match was enough, and using an XML parser is slower. I think they ended up using the XML streamer tool. After that encounter, I started using `hi` instead of `grep` for my searches for two reasons: one is convenience, as most of the time I just need an exact match, and in that case I don't want to escape special characters that would be interpreted as part of a regular expression by `grep`; and the second reason is that I know `hi` is way faster than `grep`. The difference in speed is usually not important: our everyday searches resolve in milliseconds, so unless you time it, you wouldn't be able to tell which tool ran faster. But just because I _know_ it's faster it makes me want to use it when there are no compelling reasons to do otherwise. Just to clarify, it's faster not because of some coding tricks, but because it does less.
